### ISSUE TITLE
Feature Request: One-Click Unsubscribe

### ISSUE THREAD

**Issue by gretlouise** [stcr/subscribe-to-comments-reloaded#24]
It would be great to have the features of this plugin http://wordpress.org/plugins/subscribe-to-comments-reloaded-better-unsubscribe/ added to Subscribe to Comments Reloaded, so that there's a one-click unsubscribe option available rather than just a confusing subscription management page. Thank you!

**Comment by ragelt** [comment]
+1

**Comment by IvanRF** [comment]
+1

**Comment by Reedyseth** [comment]
This feature is done and will be included on the next release.

**Comment by IvanRF** [comment]
Nice! quick question: why version numbers are dates?
[Version Numbering](https://make.wordpress.org/core/handbook/about/release-cycle/version-numbering/) | [Semantic Versioning 2.0.0](http://semver.org/)

**Comment by raamdev** [comment]
@IvanRF Date-based versions are fully compatible with Semantic Versioning 2.0. Using a date-based versioning system makes it easier to see when a particular release was done and makes more sense for things like WordPress plugins, where the plugin is constantly evolving.

**Comment by Reedyseth** [comment]
> quick question: why version numbers are dates?

This is what a @raamdev and I agree when he started to develop to the plugin. In the beginning we used the Semantic Versioning

**Comment by Reedyseth** [comment]
Thank you @raamdev for clearing this up.

**Comment by IvanRF** [comment]
@raamdev thanks for the quick response. I asked because is the first plugin (of many) that I saw using a date as version number.
WordPress plugin directory makes it easy to see when a plugin was updated with the "Last Updated" field.

**Comment by raamdev** [comment]
> WordPress plugin directory makes it easy to see when a plugin was updated with the "Last Updated" field.

@IvanRF Right, but that doesn't help developers who may be looking at the version on GitHub or inside a changelog. :-)

Date-based versioning also takes the 'guess-work' and complexity out of deciding what to use for the next version. Instead of trying to decide if the next version is `1.2`, `1.2.1`, or `2.0`, we just use the date the release was done (e.g., `150812`). 

I realize that semantic versioning defines how to increment version numbers, i.e., MAJOR.MINOR.PATCH:

> 1. MAJOR version when you make incompatible API changes,
> 1. MINOR version when you add functionality in a backwards-compatible manner, and
> 1. PATCH version when you make backwards-compatible bug fixes.

However, with WordPress plugins those points don't make as much sense: "incompatible API changes" with what? A WordPress plugin is either compatible with WordPress, or it's not. You can say a WordPress plugin is compatible with any version of WordPress greater than WordPress v4.1, but how would you indicate that with a MAJOR version number?

So using a MAJOR.MINOR.PATCH versioning system with a WordPress plugin inevitably results in breaking from the standard. You might start with 1.0, then just increment to 1.2, 1.3., and so on until you get to 1.9. But then what? Do you go to 1.10, 1.11, 1.12, and so on? Or do you jump to 2.0? 

The point being, WordPress plugins are _plugins_, i.e., they depend on the WordPress framework. In my years of developing WordPress plugins, I've found that using a date-based versioning system makes a lot more sense. :-)

**Comment by prigara** [comment]
Is there any documentation on configuring one click unsubscribe? I was trying to enable that, but with no luck. Thank you!

**Comment by Reedyseth** [comment]
Hi @prigara 

To enable the one click unsubscribe in your notifications you have to add the link on the notification message like this:

![image](https://cloud.githubusercontent.com/assets/1228711/17704091/1dc2a728-6391-11e6-8dc3-eb8a854044b1.png)

Make sure to add the tag **[oneclick_link]** on the  URL, the plugin will change that to the actual link.

Now when a new user receive a new notification they will get the One Click Unsubscribe Link.

![image](https://cloud.githubusercontent.com/assets/1228711/17704451/528554f0-6392-11e6-82c3-c727221d41ad.png)

When the click on the link they will get this message:

![image](https://cloud.githubusercontent.com/assets/1228711/17704526/81fe784c-6392-11e6-8220-672197632ca9.png)

The message that is displayed on that page is control by the one on this image, under the Notifications settings

![image](https://cloud.githubusercontent.com/assets/1228711/17704750/392582ea-6393-11e6-979a-2d4359771ef4.png)


---------------------------------------

## Don't Forget to enable the HTML Emails on the Plugin Options.

**Comment by prigara** [comment]
@Reedyseth, thanks for your reply. That worked great! I didn't know the option name - it's not listed in the description to the Notification message field.

**Comment by Reedyseth** [comment]
You're Welcome

> I didn't know the option name - it's not listed in the description to the Notification message field.

I apologize for that, my bad. On the next version is included :+1:
